### ISSUE TITLE
gnv1_nested_intel intermittent inability to match against WM baselines on wcoss/hercules/orion

### ISSUE THREAD

**Issue by zach1221** [ufs-community/ufs-weather-model#2127]
## Description

gnv1_nested_intel will frequently fail to match against baselines on WCOSS2, Orion and Hercules. The test will complete, however will not match ok, even when new baselines are created to ensure any changes to the test are captured.

### To Reproduce:

1. Log into WCOSS2
2. clone ufs-weather-model develop
3. adjust rt.conf in ufs-weather-model/tests/ so only gnv1_nested runs. 
4. run test ./rt.sh -a nems -e -l rt.conf

## Additional context

Test has failed consistently on WCOSS. Will pass occasionally on Orion and Hercules if run repeatedly.
/work2/noaa/stmp/jongkim/stmp/jongkim/FV3_RT/rt_2634567/gnv1_nested_intel

**Comment by zach1221** [comment]
@BrianCurtis-NOAA would you be able to add your recent WCOSS experiment path to the additional context section?

**Comment by zach1221** [comment]
Hi, @SamuelTrahanNOAA . When you have time, could you help us look into this issue?

**Comment by SamuelTrahanNOAA** [comment]
When did this start happening?

**Comment by BrianCurtis-NOAA** [comment]
For me on WCOSS2, a good few weeks, but it's been intermittent. The baselines are still there, but the test is disabled for WCOSS2 for now. You can re-enable it and compare against those baselines, if that helps.

**Comment by SamuelTrahanNOAA** [comment]
I need to know the specific PR that broke it.

**Comment by zach1221** [comment]
> I need to know the specific PR that broke it.

I can dig through and find out for you.

**Comment by zach1221** [comment]
> I need to know the specific PR that broke it.

WM PR#2098 is the furthest back I can find of the test failing to match on the first attempt.

**Comment by SamuelTrahanNOAA** [comment]
PR 2098 changes some NSSL microphysics code. The regression test never uses that code. It is likely that either:

1. A prior PR broke it, or
2. This problem has always been there, but we didn't notice it until recently.

Debugging a problem like this is difficult when one cannot run in debug mode. UFS nesting does not work in debug mode. It can't even compile with the GNU compiler due to syntax errors. (For example, using `.` instead of `%` to access derived type members.)

**Comment by zach1221** [comment]
> PR 2098 changes some NSSL microphysics code. The regression test never uses that code. It is likely that either:
> 
> 1. A prior PR broke it, or
> 2. This problem has always been there, but we didn't notice it until recently.
> 
> Debugging a problem like this is difficult when one cannot run in debug mode. UFS nesting does not work in debug mode. It can't even compile with the GNU compiler due to syntax errors. (For example, using `.` instead of `%` to access derived type members.)

@SamuelTrahanNOAA Are you ok with me closing this issue? It would mean gnv1_nested remains disabled on wcoss, and hercules/orion.

**Comment by SamuelTrahanNOAA** [comment]
> Are you ok with me closing this issue? It would mean gnv1_nested remains disabled on wcoss, and hercules/orion.

No. That regression test must run on all platforms. We must find out why it is failing.

**Comment by zach1221** [comment]
> > Are you ok with me closing this issue? It would mean gnv1_nested remains disabled on wcoss, and hercules/orion.
> 
> No. That regression test must run on all platforms. We must find out why it is failing.

Ok, no problem. What do you think is the best way to investigate this further without being able to run in debug?

**Comment by SamuelTrahanNOAA** [comment]
The only way I can think of is to get debug mode working with UFS FV3 nesting.

**Comment by SamuelTrahanNOAA** [comment]
I found two bugs in the nesting:

- https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/issues/328
- https://github.com/NOAA-EMC/fv3atm/issues/797

I've got fixes for both of them which I'll PR soon. It's unlikely those will fix this issue since they're both "it crashes or it runs" sorts of bugs.

**Comment by gspetro-NOAA** [comment]
@SamuelTrahanNOAA and @BrianCurtis-NOAA Is this something that you are still working on? Just looking for a status update.

**Comment by gspetro-NOAA** [comment]
> [@SamuelTrahanNOAA](https://github.com/SamuelTrahanNOAA) and [@BrianCurtis-NOAA](https://github.com/BrianCurtis-NOAA) Is this something that you are still working on? Just looking for a status update.

Any update on this issue? It seems like it may have been resolved, but I'd like to confirm.

**Comment by gspetro-NOAA** [comment]
@BrianCurtis-NOAA @jkbk2004 Do either of you know if this is still an issue? I haven't noticed it on Orion/Hercules lately; most of the time, everything passes the first time.

**Comment by BrianCurtis-NOAA** [comment]
I'll run it on WCOSS2 and let you know.

**Comment by BrianCurtis-NOAA** [comment]
running gnv1_nested intel on WCOSS2 with -c and -m was successful in comparison. We can edit that test in rt.conf for the next PR. It wouldn't be a baseline change since WCOSS2 baselines don't run that test, i'll just need to generate that baseline myself.

**Comment by gspetro-NOAA** [comment]
@BrianCurtis-NOAA I just ran on Hercules with -c and -m, and it passed the first time. Should I edit rt.conf to have this test run regularly on wcoss2 and hercules and add the change to #3047 ? The test already runs regularly on Orion without issue.

**Comment by BrianCurtis-NOAA** [comment]
> [@BrianCurtis-NOAA](https://github.com/BrianCurtis-NOAA) I just ran on Hercules with -c and -m, and it passed the first time. Should I edit rt.conf to have this test run regularly on wcoss2 and hercules and add the change to [#3047](https://github.com/ufs-community/ufs-weather-model/pull/3047) ? The test already runs regularly on Orion without issue.

Sounds good to me.
